### PR TITLE
refactor ZAP pipeline to use custom S3 Resource

### PR DESCRIPTION
**Pull request by afeld**

### CODE REVIEW

File: tasks/run-zap/task.sh [hunk 1]
```diff
@@ -25,5 +26,6 @@ done
 zap-cli shutdown
 
 echo Uploading files...
-python scripts/tools/s3upload.py
-
+# credentials are provided via environment variables
+# http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-environment
+aws s3 sync results "s3://$S3_BUCKET/results"
```
Comment (DavidEBest) [inline tasks/run-zap/task.sh]: Oh. Good call! Does this play well with versioned files in the buckets? And, can we do the reverse to sync the most recent version of the files locally? We'll need to do that for the summary file work @ctro is tackling.

Comment (afeld) [inline tasks/run-zap/task.sh]: It _shouldn't_ be any different from the behavior before...it simply takes the files passed in via the `results` folder, checks to see if they exist in the S3 bucket, then creates/updates each as necessary.

**Comment by afeld** [comment]
Holy :cow:, it works! :zap: 

screen shot 2016-03-10 at 11 41 40 pm

Ok, here are the changes:
- Consolidate the two ZAP Pipelines into two Jobs in one pipeline, meaning the Resource Type and Resource configuration can be shared
- Move S3 uploading into a custom resource: https://github.com/18F/s3-resource-simple
- Include Slack Resource Type in the Pipeline, so that Resource Type doesn't need to be set up by BOSH (i.e. we can run the Pipeline in Vagrant without any special setup
- Add/correct local Pipeline execution instructions

**Comment by DavidEBest** [comment]
@afeld This looks great! Do I need to do anything to get it running locally? I'm seeing 

```
proc_starter: ExecAsUser: system: exec of ./scripts/tasks/run-zap/task.sh: no such file or directory
```

when I run locally, and I am unsure if it is related.

**Comment by DavidEBest** [comment]
@afeld This looks great, and works locally for me. Okay to merge?

**Pull request merged**